packages/cli/src/UserManagement/PermissionChecker.ts

```diff
@@ -26,12 +26,19 @@ export class PermissionChecker {
 		// allow if all creds used in this workflow are a subset of
 		// all creds accessible to users who have access to this workflow
 
-		const workflowSharings = await Db.collections.SharedWorkflow.find({
-			relations: ['workflow'],
-			where: { workflow: { id: Number(workflow.id) } },
-		});
-
-		const workflowUserIds = workflowSharings.map((s) => s.userId);
+		let workflowUserIds = [] as string[];
+
+		// unsaved workflows have no id, so only get credentials for current user
```

Comment:
Nit: The comment refers to the else branch, so I'd place that case first. Also generally I find it more readable to have shorter consequents first.